cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo_auth.go

```diff
@@ -113,33 +114,44 @@ func (s *Server) handleRepoSecretForUpdate(
 	repoName := types.NamespacedName{Name: repo.Name, Namespace: repo.Namespace}
 	repoType := repo.Spec.Type
 
-	// handle user managed secret
+	// create/get secret
 	if hasCaRef || hasAuthRef {
+		// handle user managed secret
 		updatedSecret, err := s.validateUserManagedRepoSecret(ctx, repoName, repoType, newTlsConfig, newAuth)
 		return updatedSecret, false, true, err
-	}
 
-	// handle kubeapps managed secret
-	var isSameSecret bool
-	updatedSecret, isSameSecret, err = newSecretFromTlsConfigAndAuth(repoName, repoType, newTlsConfig, newAuth)
-	if err != nil {
-		return nil, true, false, err
-	} else if isSameSecret {
-		// Do nothing if repo auth data came redacted
-		return nil, true, false, nil
+	} else if hasCaData || hasAuthData {
+		// handle kubeapps managed secret
+		updatedSecret, isSameSecret, err := newSecretFromTlsConfigAndAuth(repoName, repoType, existingSecret, newTlsConfig, newAuth)
+		if err != nil {
+			return nil, true, false, err
+		} else if isSameSecret {
+			// Do nothing if repo auth data came fully redacted
+			return nil, true, false, nil
+		} else {
+			// secret has changed, we try to delete any existing secret
+			if existingSecret != nil {
+				if err = secretInterface.Delete(ctx, existingSecret.Name, metav1.DeleteOptions{}); err != nil {
+					log.Errorf("Error deleting existing secret: [%s] due to %v", err)
+				}
+			}
```

Comment:
I don't remember how the discussions ended, but I thought in the past we'd talked about the fact that we should not delete a secret unless it was created by kubeapps? That is, if I create a repo in Kubeapps (perhaps via the CLI, not relevant) and associate an existing secret with that repo, what should happen if I then update the repo credentials via the Kubeapps UI.

Also wondering, in what case should the error when deleting the secret here not be raised? Why is it not raised here?

Reply:
It is only deleting it because this is a secret created by kubeapps. we are not deleting secrets that kubeapps has not created.
As per the spec, there is a validation that ensures that we are not switching 'mode' between user managed and kubeapps managed secrets. So in this code path, we know this is a kubeapps managed secret.